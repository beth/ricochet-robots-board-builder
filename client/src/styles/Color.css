:local .color {
  height: 22px;
  width: 22px;
  display: grid;
  margin: auto;
}

:local .red {
  background-color: red;
}

:local .blue {
  background-color: blue;
}

:local .yellow {
  background-color: yellow;
}

:local .green {
  background-color: green;
}