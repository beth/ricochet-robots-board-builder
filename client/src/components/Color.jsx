import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Color.css';

const Color = ({
  color,
  row,
  col,
  setQuadrant,
}) => (
  <div 
    className={`${styles.color} ${styles[color.toLowerCase()]}`}
    onClick={() => setQuadrant(row, col, color)}
  />
);

Color.propTypes = {
  color: PropTypes.string.isRequired,
  row: PropTypes.number.isRequired,
  col: PropTypes.number.isRequired,
  setQuadrant: PropTypes.func.isRequired,
};

export default Color;
