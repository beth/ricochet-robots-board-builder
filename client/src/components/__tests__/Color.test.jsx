import React from 'react';
import renderer from 'react-test-renderer';
import Color from '../Color';

describe('Color Component', () => {
  it('renders correctly when given color', () => {
    const domTree = renderer.create(<Color
      color="BLUE"
      row={8}
      col={8}
      setQuadrant={() => {}}
    />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

  it('renders correctly when not given color', () => {
    const domTree = renderer.create(<Color
      color=""
      row={8}
      col={8}
      setQuadrant={() => {}}
    />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
