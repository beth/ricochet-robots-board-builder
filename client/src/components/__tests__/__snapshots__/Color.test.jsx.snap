// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component renders correctly when given color 1`] = `
<div
  className="color blue"
  onClick={[Function]}
/>
`;

exports[`Color Component renders correctly when not given color 1`] = `
<div
  className="color "
  onClick={[Function]}
/>
`;
